common/persistence/sqlMetadataPersistence.go

```diff
@@ -0,0 +1,369 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package persistence
+
+import (
+	"fmt"
+	_ "github.com/go-sql-driver/mysql" // MySQL driver
+	"github.com/jmoiron/sqlx"
+
+	"bytes"
+	"database/sql"
+	"encoding/gob"
+	workflow "github.com/uber/cadence/.gen/go/shared"
+)
+
+type (
+	sqlMetadataManager struct {
+		db *sqlx.DB
+	}
+
+	// domainRow is a flattened version of CreateDomainRequest
+	// but it also contains a DBVersion row.
+	domainRow struct {
+		DomainInfo
+		DomainConfig
+		ActiveClusterName string        `db:"active_cluster_name"`
+		Clusters          []byte        `db:"clusters"`
+		IsGlobalDomain    bool          `db:"is_global_domain"`
+		ConfigVersion     int64         `db:"config_version"`
+		FailoverVersion   int64         `db:"failover_version"`
+		DBVersion         sql.NullInt64 `db:"db_version"`
+	}
+
+	// flatUpdateDomainRequest is a flattened version of UpdateDomainRequest
+	flatUpdateDomainRequest struct {
+		DomainInfo
+		DomainConfig
+		ActiveClusterName string `db:"active_cluster_name"`
+		Clusters          []byte `db:"clusters"`
+		ConfigVersion     int64  `db:"config_version"`
+		FailoverVersion   int64  `db:"failover_version"`
+		CurrentDBVersion  int64  `db:"current_db_version"`
+		NextDBVersion     int64  `db:"next_db_version"`
+	}
+)
+
+func (m *sqlMetadataManager) ListDomain(request *ListDomainRequest) (*ListDomainResponse, error) {
+	panic("implement me")
+}
+
+const (
+	templateCreateDomainSqlQuery = `INSERT INTO domains (
+		id,
+		retention_days, 
+		emit_metric,
+		config_version,
+		name, 
+		status, 
+		description, 
+		owner_email,
+		failover_version, 
+		is_global_domain,
+		active_cluster_name, 
+		clusters, 
+db_version
+		)
+		VALUES(
+		:id,
+		:retention_days, 
+		:emit_metric,
+		:config_version,
+		:name, 
+		:status, 
+		:description, 
+		:owner_email,
+		:failover_version, 
+		:is_global_domain,
+		:active_cluster_name, 
+		:clusters,
+:db_version
+		)`
+
+	getDomainPart = `SELECT
+		id,
+		retention_days, 
+		emit_metric,
+		config_version,
+		name, 
+		status, 
+		description, 
+		owner_email,
+		failover_version, 
+		is_global_domain,
+		active_cluster_name, 
+		clusters,
+		db_version
+FROM domains
+`
+	templateGetDomainByIdSqlQuery = getDomainPart +
```

Comment:
NIT: make this name shorter. maybe calling it getDomainByIdQuery. Same below

Reply:
Not sure we can get any shorter without removing `template`. If we remove `sql` we might have a name conflict with the cassandra impl